Sources/PromotedCore/MetricsLogger.swift

```diff
@@ -144,13 +105,69 @@ public final class MetricsLogger: NSObject {
     super.init()
     self.monitor.addOperationMonitorListener(self)
   }
+}
+
+// MARK: - Ancestor IDs
+public extension MetricsLogger {
+  /// Log user ID for this session. Updated when
+  /// `startSession(userID:)` or `startSessionSignedOut()` is
+  /// called.
+  var logUserID: String? {
+    get { logUserIDProducer.currentValue }
+    set { logUserIDProducer.currentValue = newValue }
+  }
+
+  /// Log user ID for this session.
+  /// If read before the first call to `startSession*`,
+  /// returns the cached ID from the previous session from
+  /// `PeristentStore`.
+  var pendingLogUserID: String? {
```

Comment:
The name "pending" confuses me a bit.  It makes me think that something more complex is happening with the values (like committing them into a store).

It sounds like "pending" in this case is more like a "next".

Thoughts?

Reply:
For most clients, I'd imagine this behavior is what they'd expect from a field called `logUserId`.  It tries to give them what we think the logUserId should be.  I'd guess that most of the code would use this method.  Am I thinking about this wrong?

Reply:
Some of the difficulty is that I don't know the scopes for each of the variables.  I can understand the scoping is kinda crazy for this and that's why the naming conventions are hard to understand.

If I think about this as the 'next` `logUserId', I wouldn't implement this using a function called currentOrPending.

Reply:
Reading down below, pending works more like how I was originally thinking.

`currentOrPending` makes sense.  Just `pending` alone confuses me since I'd expect the next values (e.g. the next viewId).

Would most want the current `pendingLogUserId` to be the default?  I'd guess so.

Reply:
I can go with `currentOrPendingLogUserID` for the name. As for the default, over the phone I thought we aligned on the non-pending version to be the default. As in `logUserID` returns what the logger will actually log in events, and the `currentOrPending` version includes the initial ids.

Reply:
For the phone chat, the phrases that you said made sense.  I don't know the code so I have different terms for these things in my mental model.  Also, I don't remember "pending".  It's one of the reasons I wanted a video chat.  I think it'd be easier to walk through code.

When I think about this:
- Have clear defined name scope.  You can have `current` and `pending`.  Example: `current` is a value that clients can call to get the current ancestor.  E.g. viewId is not set until a View is logged.  `pending` is the next value (might be better to call it "next").
- The default (no scoped) name should be the scope that most developers think about.

Reply:
I changed the `pending` vars in the logger to `currentOrPending`. I think that of the two, the current default makes the most sense, since the actual state of the logger is easiest for developers to understand. I think of the `currentOrPending` case as the outlier.